Tools/Scripts/webkitpy/layout_tests/controllers/manager.py

```diff
@@ -726,14 +726,34 @@ def _worker_number(worker_name):
 
     def _print_expectation_line_for_test(self, format_string, test, device_type):
         test_path = test.test_path
-        line = self._expectations[device_type].model().get_expectation_line(test_path)
-        print(format_string.format(test_path,
-                                   line.expected_behavior,
-                                   self._expectations[device_type].readable_filename_and_line_number(line),
-                                   line.original_string or ''))
+        main_line = self._expectations[device_type].model().get_expectation_line(test_path)
+        if not self._options.verbose:
+            print(format_string.format(test_path,
+                                       "",
+                                       main_line.expected_behavior,
+                                       self._expectations[device_type].readable_filename_and_line_number(main_line),
+                                       main_line.original_string or ''))
+        else:
+            lines = self._expectations[device_type].model().get_expectation_lines(test_path)
+            before_main_line = True
+            decision = ""
+            for line in lines:
+                if line == main_line:
+                    decision = "-> "
+                    before_main_line = False
+                elif before_main_line:
+                    decision = "v  "
+                else:
+                    decision = "^  "
```

Comment:
Yes, it can happen. See `add_expectation_lines` in test_expectations.py, there's a test with `_already_seen_better_match()` that can discard later lines for the same file.
And it happens in practice, see the example in https://bugs.webkit.org/show_bug.cgi?id=269895  -- maybe it's a sign that one of those two lines is redundant? But in any case I think it's useful information.

I didn't see tests for `get_expectation_line`, that's why I lazily didn't bother testing `get_expectation_lines` 😅 But yeah, it could clarify what's supposed to happen.

I don't quite understand your last paragraph with the `lines[-1]`.
In non-verbose mode, we'll always use a literal "" (at line 732).
With verbose, we'll always have a 4-character decision containing either "v", "->", or "^".

Reply:
I've added some tests for the basic functionality, the last one shows how the "main" expectation line may not necessarily be the very last one (in that case, a later override of the whole directory, but with the same expected result.)